sqlitecluster/SQLite.h

```diff
@@ -192,6 +198,8 @@ class SQLite {
     // Enable/disable SQL statement tracing.
     static atomic<bool> enableTrace;
 
+    static atomic<sqlite3_stmt*> _lastQuery;
```

Comment:
Like above, this should probably be a string and doesn't need to be atomic, but specifically it needs to *not* be `static`. Otherwise there's no way for each `SQLite` object to know it's looking at the right value.

Reply:
mh, the only way to get a `sqlite3_stmt` is in `_sqliteTraceCallback` which is static. Is there another way to get it?

Reply:
We register the callback function here: https://github.com/Expensify/Bedrock/blob/master/sqlitecluster/SQLite.cpp#L141

You can see the fourth argument to `sqlite3_trace_v2` is `this`, a pointer to the current SQLite object.

So, based on the (somewhat confusing) docs here: https://www.sqlite.org/c3ref/trace_v2.html the `c` argument to `_sqliteTraceCallback` here: https://github.com/Expensify/Bedrock/blob/master/sqlitecluster/SQLite.cpp#L233 is the same as `this`, so you can do:

```
c->_lastQuery
```

To access the last query for this particular object.